test/python/transpiler/test_optimize_1q_decomposition.py

```diff
@@ -415,7 +415,21 @@ def test_optimize_u3_to_u2(self):
         passmanager.append(BasisTranslator(sel, basis))
         passmanager.append(Optimize1qGatesDecomposition(basis))
         result = passmanager.run(circuit)
+        self.assertEqual(expected, result)
 
+    def test_y_simplification_rz_sx_x(self):
+        """Test that a y gate gets decomposed to x-zx with ibmq basis."""
+        qc = QuantumCircuit(1)
+        qc.y(0)
+        basis = ["id", "rz", "sx", "x", "cx"]
+        passmanager = PassManager()
+        passmanager.append(BasisTranslator(sel, basis))
+        passmanager.append(Optimize1qGatesDecomposition(basis))
+        passmanager.append(Optimize1qGatesDecomposition(basis))
```

Comment:
Is the second pass of `Optimize1qGatesDecomposition` needed?

Reply:
not strictly, a single execution results in `[Rz(pi/2), X, Rz(3pi/2)]` The second execution further reduces this to `[X, Rz(pi)]` but we can just test for the first decomposition result as the expected. I'll update this now

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/5937/commits/de3cf6b7f088693ddd9dcc1f5107c984bb0c90c6

Reply:
So this is failing on precision differences on different platforms, so I'm going to revert this change and just leave the double calls (I'll add a comment)